src/providers/node/mod.rs

```diff
@@ -203,9 +219,25 @@ impl Provider for NodeProvider {
         // Start
         let start = NodeProvider::get_start_cmd(app, env)?.map(StartPhase::new);
 
-        let mut plan = BuildPlan::new(&vec![setup, install, build], start);
-        plan.add_variables(NodeProvider::get_node_environment_variables());
+        let mut phases = vec![setup, install, build];
+        if let Some(caddy) = SpaProvider::caddy_phase(app, env) {
+            phases.insert(1, caddy); // insert after setup and before build
+        }
 
+        let mut plan = BuildPlan::new(&phases, start);
+        if SpaProvider::caddy_phase(app, env).is_some() {
```

Comment:
Suggestion: We call this a few times in the file. We should call it once and save it to a variable for reuse.